src/jsdoc/react/react-parser.ts

```diff
@@ -0,0 +1,218 @@
+import doctrine from 'doctrine';
+import exampleTagParser from '../example-tag-parser';
+import { PathOsBased } from '../../utils/path';
+import { pathNormalizeToLinux } from '../../utils';
+import logger from '../../logger/logger';
+import { Doclet } from '../types';
+
+const reactDocs = require('react-docgen');
+
+function formatTag(tag: Record<string, any>): Record<string, any> {
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  delete tag.title;
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  if (!tag.type) return tag;
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  let formattedType = doctrine.type.stringify(tag.type);
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  if (tag.type.type === doctrine.type.Syntax.TypeApplication) {
+    // Doctrine adds a dot after the generic type for historical reasons.
+    // see here for more info: https://github.com/eslint/doctrine/issues/185
+    formattedType = formattedType.replace('.<', '<');
+  }
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  if (tag.type.type === doctrine.type.Syntax.OptionalType) {
+    // Doctrine shows an optional type with a suffix `=` (e.g. `string=`), we prefer the more
+    // common syntax `?` (e.g. `string?`)
+    formattedType = formattedType.replace('=', '?');
+  }
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  tag.type = formattedType;
+
+  return tag;
+}
+
+function extractDataRegex(doc: string, doclets: Array<Doclet>, filePath: PathOsBased) {
```

Comment:
This function exist twice, here and in the jsdoc parse. consolidate them

Reply:
I fixed this